src/main/java/com/google/vinet/servlets/LoginStatusServlet.java

```diff
@@ -0,0 +1,62 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.vinet.servlets;
+
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.users.UserService;
+import com.google.gson.Gson;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns the login status of the user. */
+@WebServlet("/login-status")
+public class LoginStatusServlet extends HttpServlet {
+  private UserService userService;
+
+  /**
+   * Construct a LoginStatusServlet that depends on the default implementaion of UserService.
+   */
+  public LoginStatusServlet() {
+    this.userService = UserServiceFactory.getUserService();
+  }
+
+  /**
+   * Construct a LoginStatusServlet that depends on the provided implementaion of UserService.
+   * @param userService The implementation of UserService the LoginStatusServlet will depend on.
+   */
+  public LoginStatusServlet(UserService userService) {
+    this.userService = userService;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    if (response == null) {
+      throw new IllegalArgumentException("response cannot be null");
+    }
+
+    if (request == null) {
+      throw new IllegalArgumentException("request cannot be null");
+    }
```

Comment:
Will the frontend just not get a response if an exception is thrown?

Reply:
see: https://github.com/googleinterns/step226-2020/pull/6/files/dcf8e6d89c588aea0820764ac1210408dd6ae1db#r477188380